kornia/color/zca.py

```diff
@@ -0,0 +1,208 @@
+
+        [1] `Stanford PCA & ZCA whitening tutorial <http://ufldl.stanford.edu/tutorial/unsupervised/PCAWhitening/>`_
+
+    """
+
+    def __init__(self, eps: float = 1e-7, biased: bool = False, detach_transforms: bool = True) -> None:
+
+        super(ZCAWhitening, self).__init__()
+
+        self.eps: float = eps
+        self.biased: bool = biased
+        self.detach_transforms: bool = detach_transforms
+        self.fitted = False
+
+    def fit(self, x: torch.Tensor):  # type: ignore
+        r"""
+
+        Fits ZCA whitening matrices to the data.
+
+        args:
+
+            x (torch.Tensor): Input data
+
+        returns:
+            ZCAWhiten: returns a fitted ZCAWhiten object instance.
+        """
+
+        T, mean, T_inv = zca_whitening_transforms(x, self.eps, self.biased, compute_inv=True)
+
+        self.mean: torch.Tensor = mean
+        self.T: torch.Tensor = T
+        self.T_inv: torch.Tensor = T_inv
+
+        if self.detach_transforms:
+            self.mean = self.mean.detach()
+            self.T = self.T.detach()
+            self.T_inv = self.T_inv.detach()
+
+        self.fitted = True
+
+        return self
+
+    def forward(self, x: torch.Tensor, include_fit: bool = False) -> torch.Tensor:  # type: ignore
+        r"""
+
+        Applies the whitening transform to the data
+
+        args:
+
+            x (torch.Tensor): Input data
+            include_fit (bool): Indicates whether to fit the data as part of the forward pass
+
+        returns:
+
+            torch.Tensor : The transformed data
+
+        """
+
+        if include_fit:
+            self.fit(x)
+
+        if not self.fitted:
+            raise RuntimeError("Needs to be fitted first before running. Please call fit or set include_fit to True.")
+
+        num_features: int = reduce(lambda a, b: a * b, x.size()[1::])
+
+        x_flat: torch.Tensor = torch.reshape(x, (-1, num_features))
+
+        y = (x_flat - self.mean).mm(self.T)
+
+        y = y.reshape(x.size())
+
+        return y
+
+    def inverse_transform(self, x: torch.Tensor) -> torch.Tensor:
+        r"""
+
+        Applies the inverse transform to the whitened data.
+
+        args:
+            x (torch.Tensor): Whitened data
+            include_fit (bool): Indicates whether to fit the data as part of the forward pass
+
+        returns:
+            torch.Tensor: original data
+
+
+
+        """
+
+        if not self.fitted:
+            raise RuntimeError("Needs to be fitted first before running. Please call fit or set include_fit to True.")
+
+        num_features: int = reduce(lambda a, b: a * b, x.size()[1::])
+
+        x_flat: torch.Tensor = torch.reshape(x, (-1, num_features))
+
+        y: torch.Tensor = (x_flat).mm(self.T_inv) + self.mean
+
+        y = y.reshape(x.size())
+
+        return y
+
+
+def zca_whitening_transforms(inp: torch.Tensor, eps: float = 1e-7,
+                             biased: bool = False, compute_inv: bool = False) -> Tuple[torch.Tensor, ...]:
```

Comment:
I would rename to `kornia.zca_mean` or similar, in order to be consistent to pytorch functions like `torch.std_mean`
https://pytorch.org/docs/stable/torch.html#torch.std_mean

I propose this signtature:
```python
def zca_mean(input: torch.Tensor, unbiased: bool = True, return_inverse: bool = False, eps: float = 1e-7) -> Tuple[torch.Tensor, ...]:
```

Reply:
Sure I can rename it.